### DISCUSSION TITLE
Increasing Java memory allocation and changing the stack size on Github Actions

### DISCUSSION THREAD

**Discussion by stefan-jung** [dita-ot/dita-ot#4333] [Q&A] [answered]
Do you know if and how I can set [increase Java memory](https://www.dita-ot.org/dev/topics/increasing-the-jvm) and change the stack size on Github Actions?

In Oxygen XML it has helped (thanks a lot again, @raducoravu) to change the JVM arguments. But I could not figure out how to do that on Github Actions. Here the builds are also failing with a `java.lang.StackOverflowError`. I know, this is not directly dita-ot related, but I do not know where else I should ask this question.

![image](https://github.com/dita-ot/dita-ot/assets/3266567/74dcd32d-0aa0-49dc-8c0e-d5419a8a6f1f)

```
transform.topic2fo.main:
 [pipeline] Loading stylesheet file:/home/runner/work/com.dometic.dita-ot-pdf/com.dometic.dita-ot-pdf/dita-ot/plugins/com.dometic.dita-ot-pdf/xsl/fo/topic2fo_shell_dometic_fop.xsl
Error: The following error occurred while executing this line:
/home/runner/work/com.dometic.dita-ot-pdf/com.dometic.dita-ot-pdf/dita-ot/plugins/org.dita.pdf2/build.xml:280: java.lang.StackOverflowError
Error: Process completed with exit code 1.
```

**Comment by infotexture** [comment]
@stefan-jung You might want to try [larger runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners).

**Reply by stefan-jung** [reply]
Thanks, @infotexture, did not know that this exists.

**Comment by stefan-jung** [comment, marked as answer]
Setting this has helped:

```yaml
env:
  ANT_OPTS: '-Xmx12000m -Xss10m'
```
